app.py

```diff
@@ -1,7 +1,141 @@
-from flask import Flask, render_template
-app = Flask(__name__)
+"""
+A Flask application for snapcraft.io.
+
+The web frontend for the snap store.
+"""
+
+import flask
+import requests
+import requests_cache
+import datetime
+import json
+import humanize
+from dateutil import parser
+
+
+app = flask.Flask(__name__)
+uncached_session = requests.Session()
+cached_session = requests_cache.CachedSession(expire_after=60)
+request_timeout = 2
+snap_details_url = (
+    "https://api.snapcraft.io/api/v1/snaps/details/{snap_name}"
+    "?channel=stable"
+)
 
 
 @app.errorhandler(404)
-def page_not_found(e):
-    return render_template('404.html'), 404
+def page_not_found(error):
+    """
+    For 404 pages, display the 404.html template,
+    passing through the error description.
+    """
+
+    return flask.render_template(
+        '404.html', description=error.description
+    ), 404
+
+
+@app.route('/<snap_name>/')
+def snap_details(snap_name):
+    """
+    A view to display the snap details page for specific snaps.
+
+    This queries the snapcraft API (api.snapcraft.io) and passes
+    some of the data through to the snap-details.html template,
+    with appropriate sanitation.
+    """
+
+    query_headers = {
+        'X-Ubuntu-Series': '16',
+    }
+
+    response = _get_from_cache(
+        snap_details_url.format(snap_name=snap_name),
+        headers=query_headers
+    )
+
+    if response.status_code >= 400:
+        message = (
+            'Failed to get snap details for {snap_name}'.format(**locals())
+        )
+
+        if response.status_code == 404:
+            message = 'Snap not found: {snap_name}'.format(**locals())
+
+        flask.abort(response.status_code, message)
+
+    snap_data = json.loads(response.text)
+
+    context = {
+        # Always available
+        'name': snap_data['title'],
+        'api_error': response.old_data_from_error,
+        'version': snap_data['version'],
+        'revision': snap_data['revision'],
+        'filesize': humanize.naturalsize(snap_data['binary_filesize']),
+        'download_url': snap_data['download_url'],
+        'publisher': snap_data['publisher'],
+        'screenshot_urls': snap_data['screenshot_urls'],
+        'prices': snap_data['prices'],
+        'support_url': snap_data.get('support_url'),
+        'last_updated': (
+            humanize.naturaldate(
+                parser.parse(snap_data.get('last_updated'))
+            )
+        ),
+
+        # May be available
+        'summary': snap_data.get('summary'),
+        'description': snap_data.get('description'),
+        'icon_url': snap_data.get('icon_url'),
```

Comment:
These fields are always present (i.e available in the response payload), they just happen to be eventually empty. No need for `.get()`s.